server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -0,0 +1,924 @@
+        log.info("Group {}, deleteGroup.", group);
+
+        return store.deleteGroup(namespace, group)
+                    .whenComplete((r, e) -> {
+                        if (e == null) {
+                            log.info("Group {}, group deleted", group);
+                        } else {
+                            log.warn("Group {}, group delete failed with error", group, e);
+                        }
+                    });
+    }
+
+    /**
+     * List of compressions used for encoding in the group.
+     *
+     * @param namespace namespace for which the request is scoped to.
+     * @param group     Name of group.
+     * @return CompletableFuture that holds list of compressions used for encoding in the group.
+     */
+    public CompletableFuture<List<String>> getCodecTypes(String namespace, String group) {
+        Preconditions.checkArgument(group != null);
+        log.info("Group {}, getCodecTypes.", group);
+
+        return store.getCodecTypes(namespace, group)
+                    .whenComplete((r, e) -> {
+                        if (e == null) {
+                            log.info("Group {}, codecTypes = {}", group, r);
+                        } else {
+                            log.warn("Group {}, getcodecTypes failed with error", group, e);
+                        }
+                    });
+    }
+
+    /**
+     * Method to add codecType to a group. Adding codecType is idempotent. CodecTypes need to be registered before
+     * using them in encoding id related Apis.
+     *
+     * @param namespace namespace for which the request is scoped to.
+     * @param group     group.
+     * @param codecType codec type to add.
+     * @return CompletableFuture which when completed successfully will indicate successful addition of codecType to group.
+     */
+    public CompletableFuture<Void> addCodecType(String namespace, String group, String codecType) {
+        Preconditions.checkArgument(group != null);
+        Preconditions.checkArgument(codecType != null);
+
+        log.info("Group {}, addCodecType {}.", group, codecType);
+
+        return store.addCodecType(namespace, group, codecType)
+                    .whenComplete((r, e) -> {
+                        if (e == null) {
+                            log.info("Group {}, addCodecType {} successful", group, codecType);
+                        } else {
+                            log.warn("Group {}, addCodecType {} failed with error", group, codecType, e);
+                        }
+                    });
+
+    }
+
+    private boolean validateRules(SerializationFormat serializationFormat, Compatibility newRules) {
+        switch (serializationFormat) {
+            case Avro:
+                return true;
+            case Protobuf:
+            case Json:
+            case Custom:
+            case Any:
+                return newRules.getType().equals(Compatibility.Type.AllowAny) || newRules.getType().equals(Compatibility.Type.DenyAll);
```

Comment:
While this is a single statement, its intent is not clear in the context of this method. I think you should extract this out an explaining method with a name that makes it's intent immediately apparent. 

Or, alternatively, renaming the method param `newRules` to something like `compatibilityPolicy` or the like might help.

Reply:
renamed newRules  to compatibility